develop/jsapi/search_with_widget/lab.md

```diff
@@ -0,0 +1,89 @@
+###Search with Widget
+
+In this lab you will add a search widget and search against a feature layer. The widget performs context-sensitve search as you type and then it will zoom to and highlight the feature selected. You can also format the data in the popup that appears. 
+
+In this lab it will search against the neighborhood polygon layer but you can point to any hosted feature layer you want.
+
+1. Click [create_starter_map/index.html](../create_starter_map/index.html) and copy the contents to a new [jsbin.com](http://jsbin.com).
+
+2. In `JSBin` > `HTML`, update the `require` statement and function definition:
+
+    ```javascript
+    require([
+      "esri/Map",
+      "esri/views/MapView",
+      "esri/widgets/Search",
+      "esri/layers/FeatureLayer",
+      "dojo/domReady!"
+    ], function(Map, MapView, Search, FeatureLayer) {
+      ... 
+    ```
+
+3. Create the Search widget and add it to the `top-right` of the UI.
+
+    ```javascript
+    ], function(Map, MapView, Search, FeatureLayer) {
+
+      var map = new Map({
+        basemap: "dark-gray"
+      });
+
+      var view = new MapView({
+        container: "viewDiv",
+        map: map,
+        center: [-122.68, 45.52], // lon, lat
+        zoom: 10
+      });
+
+      var searchWidget = new Search({
+        view: view,
+        allPlaceholder: "Neighborhood e.g. Downtown"
+      });
+
+      searchWidget.startup();
+
+      view.ui.add(searchWidget, {
+        position: "top-right"
+      });
+    ```
+
+    At this point, the map will allow you to search against the default ArcGIS Online Geocoding Service. Give it a go. You can enter an address or point of interest (like `Providence Park` or `PDX`) or a geography (like `Oregon` or `USA`).
+
+4. Now configure the Search Widget to just search against the Neighborhoods Feature Service. Insert the following code directly before the `search.startup()` line added above.
+
+    ```javascript
+      var searchWidget = new Search({
```

Comment:
You don't add the searchWidget again. Should you remove these first few lines and start with the `var sources = [];`?